### ISSUE TITLE
Package version numbers

### ISSUE THREAD

**Issue by warpech** [Starcounter/Warehouse#102]
By default, Visual Studio creates four segment versioning, like: `1.0.0.0`. This is a traditional Microsoft's `AssemblyVersion` scheme, where the digits represent: `MAJOR.MINOR.BUILD.REVISION`. 

More about it here:

- https://blogs.msdn.microsoft.com/jjameson/2009/04/03/best-practices-for-net-assembly-versioning/
- https://msdn.microsoft.com/en-us/library/51ket42z(v=vs.110).aspx

Every modern stack in the world follows semver (`MAJOR.MINOR.PATCH`), which does not include a build number:

- NuGet 1.6-3.x follows SemVer 1.0.0 (see http://blog.nuget.org/20140924/supporting-semver-2.0.0.html)

- Cargo follows SemVer 2.0.0 (see http://doc.crates.io/specifying-dependencies.html)

- npm uses SemVer 2.0.0 (see https://docs.npmjs.com/getting-started/semantic-versioning)

Could we adapt semver instead of Microsoft's nubmering? This brings a benefit of being supported by many version comparison libraries.

Also, what if someone builds apps using C# on Linux or NodeJS in future? Microsoft's scheme seems an unnatural fit then.

**Comment by warpech** [comment]
I just check that this is possible, for example:

Put this info in **AssemblyInfo.cs**: `[assembly: AssemblyFileVersion("5.0.1")]` 

or provide it in the dialog with leaving the last field empty: 
![2016-11-16 15_14_56-assembly information](https://cloud.githubusercontent.com/assets/566463/20350502/7c0598ba-ac0f-11e6-99e9-91288ce9ad5a.png)

After changing AssemblyInfo, you need to **rebuild** the project.

**Comment by andwah** [comment]
> Could we adapt semver instead of Microsoft's numbering? This brings a benefit of being supported by many version comparison libraries.

It's up to the app developer to choose what version 'system' he wants.

The warehouse just stores the version as a 'string', it uses the version date to get the order/latest.

**Comment by warpech** [comment]
Good point. In our sample apps we should follow one pattern to appear consistent.
